synapse/storage/presence.py

```diff
@@ -112,6 +112,22 @@ def _update_presence_txn(self, txn, stream_id, presence_states):
                 args
             )
 
+    def get_all_presence_updates(self, last_id, current_id):
+        def get_all_presence_updates_txn(txn):
+            sql = (
+                "SELECT stream_id, user_id, state, last_active_ts,"
+                " last_federation_update_ts, last_user_sync_ts, status_msg,"
+                " currently_active"
```

Comment:
Hmm, yeah I wasn't sure if they were going to be useful, I didn't think there was any harm in including them though.